flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+    } else {
+      return new MaterialButton(onPressed: () {}, child: new Text('Toggle Two'));
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text("Sample App"),
+      ),
+      body: new Center(
+        child: _getToggleChild(),
+      ),
+      floatingActionButton: new FloatingActionButton(
+        onPressed: _toggle,
+        tooltip: 'Update Text',
+        child: new Icon(Icons.update),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+
+**In Android, I can Animate a view by View.animate(), how can I do that to a Widget?**
+In Flutter, animating widgets can be done via the animation library.
+
+In Android you would either create animations via XML or call the .animate() 
+property on Views, in Flutter you can wrap widgets inside an Animation.
+
+Like Android, in Flutter you have an AnimationController and a 
+Interpolator which is an extension of the Animation class, for example a 
+CurvedAnimation. You pass the controller and Animation into an AnimationWidget 
+and tell the controller to start the animation.
+
+Let's take a look at how to write a FadeTransition that will Fade in a logo 
+when you press
+
+<!-- skip -->
+{% prettify dart %}
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(new FadeAppTest());
+}
+
+class FadeAppTest extends StatelessWidget {
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Fade Demo',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyFadeTest(title: 'Fade Demo'),
+    );
+  }
+}
+
+class MyFadeTest extends StatefulWidget {
+  MyFadeTest({Key key, this.title}) : super(key: key);
+  final String title;
+  @override
+  _MyFadeTest createState() => new _MyFadeTest();
+}
+
+class _MyFadeTest extends State<MyFadeTest> with TickerProviderStateMixin {
+  AnimationController controller;
+  CurvedAnimation curve;
+
+  @override
+  void initState() {
+    controller = new AnimationController(duration: const Duration(milliseconds: 2000), vsync: this);
+    curve = new CurvedAnimation(parent: controller, curve: Curves.easeIn);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text(widget.title),
+      ),
+      body: new Center(
+          child: new Container(
+              child: new FadeTransition(
+                  opacity: curve,
+                  child: new FlutterLogo(
+                    size: 100.0,
+                  )))),
+      floatingActionButton: new FloatingActionButton(
+        tooltip: 'Fade',
+        child: new Icon(Icons.brush),
+        onPressed: () {
+          controller.forward();
+        },
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+ See https://flutter.io/widgets/animation/ and https://flutter.io/tutorials/animation for more specific details.
```

Comment:
do you have to markup these two URLs , in order for them to be links noticed by Markdown?

Reply:
done!